actors/miner/tests/fip0081_initial_pledge.rs

```diff
@@ -0,0 +1,232 @@
+        fn $name() -> $ret_type {
+            $value
+        }
+    };
+}
+
+my_const!(epoch_target_reward, TokenAmount, zero());
+my_const!(qa_sector_power, StoragePower, StoragePower::from(1u64 << 36));
+my_const!(network_qa_power, StoragePower, StoragePower::from(1u64 << 10));
+my_const!(power_rate_of_change, StoragePower, StoragePower::from(1u64 << 10));
+my_const!(
+    reward_estimate,
+    FilterEstimate,
+    FilterEstimate::new(epoch_target_reward().atto().clone(), zero())
+);
+my_const!(
+    power_estimate,
+    FilterEstimate,
+    FilterEstimate::new(network_qa_power(), power_rate_of_change())
+);
+my_const!(circulating_supply, TokenAmount, TokenAmount::from_whole(1));
+
+// Pre-ramp where 'baseline power' dominates
+#[test]
+fn initial_pledge_pre_ramp_negative() {
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        -100,
+        100,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1500).div_floor(10000),
+        initial_pledge
+    );
+}
+
+#[test]
+fn initial_pledge_zero_ramp_duration() {
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        0,
+        0,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1950).div_floor(10000),
+        initial_pledge
+    );
+
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        10,
+        0,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1950).div_floor(10000),
+        initial_pledge
+    );
+}
+
+// Pre-ramp where 'baseline power' dominates
+#[test]
+fn initial_pledge_pre_ramp() {
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        0,
+        100,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1500).div_floor(10000),
+        initial_pledge
+    );
+}
+
+// On-ramp where 'baseline power' (85%).
+#[test]
+fn initial_pledge_on_ramp_mid() {
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        50,
+        100,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1725).div_floor(10000),
+        initial_pledge
+    );
+}
+
+// On-ramp where 'baseline power' (85%).
```

Comment:
This comment is wrong, this is past ramp

Reply:
Fixed.